w3lib/url.py

```diff
@@ -47,13 +50,13 @@ def urljoin_rfc(base, ref, encoding='utf-8'):
 
     str_base = unicode_to_str(base, encoding)
     str_ref = unicode_to_str(ref, encoding)
-    return moves.urllib.parse.urljoin(str_base, str_ref)
+    return urljoin(str_base, str_ref)
 
 _reserved = b';/?:@&=+$|,#' # RFC 3986 (Generic Syntax)
 _unreserved_marks = b"-_.!~*'()" # RFC 3986 sec 2.3
 _safe_chars = _ALWAYS_SAFE_BYTES + b'%' + _reserved + _unreserved_marks
 
-def safe_url_string(url, encoding='utf8'):
+def safe_url_string(url, encoding='utf8', path_encoding='utf8'):
     """Convert the given url into a legal URL by escaping unsafe characters
     according to RFC-3986.
 
```

Comment:
I think the docstring below should also be updated: the unicode url is not converted to str using given encoding. I'd suggest something like (following the summary from the PR):
```
If a bytes url is given, it is first converted to str using the given
encoding (which defaults to 'utf-8'). 'utf-8' encoding is used for
URL path component (unless overriden by path_encoding), and given
encoding is used for query string or form data.
When passing a encoding, you should use the encoding of the
original page (the page from which the url was extracted from).
```